src/pykeen/models/base.py

```diff
@@ -398,7 +398,7 @@ def predict_tails(
     def _novel(self, h, r, t) -> bool:
         """Return if the triple is novel with respect to the training triples."""
         triple = torch.tensor(data=[h, r, t], dtype=torch.long, device=self.device).view(1, 3)
-        return (triple == self.triples_factory.mapped_triples).all(dim=1).any().item()
+        return (triple == self.triples_factory.mapped_triples.to(self.device)).all(dim=1).any().item()
```

Comment:
Since `mapped_triples` is not already on the device, it does not make much sense to move all triples to the device, just to check whether one of them equals *one* triple of interest. I'll check the code and push something later 🙂

Reply:
Would then `triple = torch.tensor(data=[h, r, t], dtype=torch.long, device=self.triples_factory.mapped_triples.device).view(1, 3)` not be the easier solution?

Reply:
Yes. The main point I fixed is that we call `_novel` for every triple independently, although `(h,r)` or `(r,t)` stay fixed. So we can save a lot when we filter only once.

Reply:
Ok, makes sense :)
Here, I only wanted to fix the problem ;)

Reply:
Fair point. I moved the changes to a separate PR, #51 , and reverted the changes I made to this branch.

However, the new PR supersedes this PR, since it makes the problem obsolete the PR is trying to fix 😉

Reply:
We can just close this PR :)